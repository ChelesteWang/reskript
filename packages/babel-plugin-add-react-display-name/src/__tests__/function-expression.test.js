const runSuite = require('./suite');

runSuite('function-expression');
