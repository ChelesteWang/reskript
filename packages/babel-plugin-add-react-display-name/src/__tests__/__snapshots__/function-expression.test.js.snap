// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function expression anonymous function assigned to object property 1`] = `
"// anonymous function assigned to object property

/* eslint-disable */
import { Component } from 'react';
const global = {
  namespace: {}
};

global.namespace.Foo = function () {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};

global.namespace.Foo.displayName = \\"Foo\\";"
`;

exports[`function expression anonymous function assigned to variable 1`] = `
"// anonymous function assigned to variable

/* eslint-disable */
const Foo = function () {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};

Foo.displayName = \\"Foo\\";"
`;

exports[`function expression as class member 1`] = `
"function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

// as class member

/* eslint-disable */
class Foo {
  constructor() {
    _defineProperty(this, \\"Bar\\", function () {
      return /*#__PURE__*/React.createElement(\\"div\\", null);
    });
  }

}"
`;

exports[`function expression as named export 1`] = `
"// as named export

/* eslint-disable */
export const Foo = function () {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};
Foo.displayName = \\"Foo\\";"
`;

exports[`function expression as named export with function name 1`] = `
"// as named export with function name

/* eslint-disable */
export const Foo = function Bar() {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};
Foo.displayName = \\"Bar\\";"
`;

exports[`function expression assignment to variable 1`] = `
"// assignment to variable

/* eslint-disable */
let Foo;

Foo = function () {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};

Foo.displayName = \\"Foo\\";"
`;

exports[`function expression function in iife 1`] = `
"// function in iife

/* eslint-disable */
(function Foo() {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
})();"
`;

exports[`function expression named function assigned to object property 1`] = `
"// named function assigned to object property

/* eslint-disable */
import { Component } from 'react';
const global = {
  namespace: {}
};

global.namespace.Foo = function Bar() {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};

global.namespace.Foo.displayName = \\"Bar\\";"
`;

exports[`function expression nested in child scope as a variable 1`] = `
"// nested in child scope as a variable

/* eslint-disable */
const factory = () => {
  const Foo = function () {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  };

  Foo.displayName = \\"Foo\\";
  return Foo;
};"
`;

exports[`function expression nested in child scope as an assignment 1`] = `
"// nested in child scope as an assignment

/* eslint-disable */
const initialize = () => {
  window.Foo = function () {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  };

  window.Foo.displayName = \\"Foo\\";
};"
`;

exports[`function expression returned from function 1`] = `
"// returned from function

/* eslint-disable */
const factory = () => function () {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};"
`;
